src/v0/destinations/fb_custom_audience/util.js

```diff
@@ -126,43 +127,77 @@ const ensureApplicableFormat = (userProperty, userInformation) => {
+    // for MOBILE_ADVERTISER_ID, MADID,EXTERN_ID hashing is not required ref: https://developers.facebook.com/docs/marketing-api/audiences/guides/custom-audiences#hash
+    if (tmpUpdatedProperty) {
+      tmpUpdatedProperty = `${tmpUpdatedProperty}`;
+      dataElement.push(sha256(tmpUpdatedProperty));
+    } else {
+      dataElement.push(null);
+    }
+  } else if (!tmpUpdatedProperty && (eachProperty === 'MADID' || eachProperty === 'EXTERN_ID')) {
+    dataElement.push(null);
+  } else {
+    dataElement.push(tmpUpdatedProperty);
+  }
+  return dataElement;
+};
+
 // Function responsible for making the data field without payload object
 // Based on the "isHashRequired" value hashing is explicitly enabled or disabled
-const prepareDataField = (userSchema, userUpdateList, isHashRequired, disableFormat) => {
+const prepareDataField = (
+  userSchema,
+  userUpdateList,
+  isHashRequired,
+  disableFormat,
+  destinationId,
+) => {
   const data = [];
-  let updatedProperty;
-  let dataElement;
+  let nullEvent = true; // flag to check for bad events (all user properties are null)
+
   userUpdateList.forEach((eachUser) => {
-    dataElement = [];
+    let dataElement = [];
+    let nullUserData = true; // flag to check for bad event (all properties are null for a user)
+
     userSchema.forEach((eachProperty) => {
       const userProperty = eachUser[eachProperty];
-      if (isHashRequired) {
-        if (!disableFormat) {
-          // when user requires formatting
-          updatedProperty = ensureApplicableFormat(eachProperty, userProperty);
-        } else {
-          // when user requires hashing but does not require formatting
-          updatedProperty = userProperty;
-        }
-      } else {
-        // when hashing is not required
-        updatedProperty = userProperty;
+      let updatedProperty = userProperty;
+
+      if (isHashRequired && !disableFormat) {
+        updatedProperty = ensureApplicableFormat(eachProperty, userProperty);
       }
-      if (isHashRequired && eachProperty !== 'MADID' && eachProperty !== 'EXTERN_ID') {
-        // for MOBILE_ADVERTISER_ID, MADID,EXTERN_ID hashing is not required ref: https://developers.facebook.com/docs/marketing-api/audiences/guides/custom-audiences#hash
-        if (updatedProperty) {
-          updatedProperty = `${updatedProperty}`;
-          dataElement.push(sha256(updatedProperty));
-        } else {
-          dataElement.push(null);
-        }
-      } else {
-        dataElement.push(updatedProperty);
+
+      dataElement = getUpdatedDataElement(
+        dataElement,
+        isHashRequired,
+        eachProperty,
+        updatedProperty,
+      );
+
+      if (dataElement[dataElement.length - 1]) {
+        nullUserData = false;
+        nullEvent = false;
       }
     });
+
+    if (nullUserData) {
+      stats.increment('fb_custom_audience_event_having_all_null_field_values_for_a_user', {
```

Comment:
[Q1] Will the stat add value if we know which field is null ?
[Q2] Are the user-fields going to a lot ? I was asking in terms of cardinality

Reply:
>Are the user-fields going to a lot ?

The intention is to know if these user-fields are fixed but the numbers are high or if the user-fields are very random. With that we can estimate the cardinality

Reply:
1. We just want to get the count of the event in which all the fields are null, doesn't matter which field is null here.
Didn't get the second question.

Reply:
>We just want to get the count of the event in which all the fields are null, doesn't matter which field is null here.

I understand, but if I have a label which says that this field that is sent is `null`, would it add value ? -- This is my question

Second question was, how many user-fields would be sent ? Can it be randomly sent by user or is it a fixed set of values that should be set as part of the payload ?!